oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisOperatingSystem.java

```diff
@@ -129,32 +131,43 @@ public OSProcess getProcess(int pid) {
 
     @Override
     public List<OSProcess> queryDescendantProcesses(int parentPid) {
-        Set<String> descendantPids = new HashSet<>();
-        addChildrenToDescendantSet(Integer.toString(parentPid), descendantPids, true);
+        Set<String> descendantPids = getChildrenOrDescendants(Integer.toString(parentPid), true);
         if (descendantPids.isEmpty()) {
             return Collections.emptyList();
         }
         return getProcessListFromPS("ps -o s,pid,ppid,user,uid,group,gid,nlwp,pri,vsz,rss,etime,time,comm,args -p "
                 + String.join(",", descendantPids), -1);
     }
 
-    private static void addChildrenToDescendantSet(String parentPid, Set<String> descendantPids, boolean recurse) {
-        // Get list of children
+    private static List<OSProcess> queryAllProcessesFromPS() {
+        return getProcessListFromPS("ps -eo s,pid,ppid,user,uid,group,gid,nlwp,pri,vsz,rss,etime,time,comm,args", -1);
+    }
+
+    private static Set<String> getChildren(String parentPid) {
         Set<String> childPids = new HashSet<>();
         for (String s : ExecutingCommand.runNative("pgrep -P " + parentPid)) {
             String pid = s.trim();
-            if (!pid.equals(parentPid) && !descendantPids.contains(pid)) {
+            if (!pid.equals(parentPid)) {
                 childPids.add(pid);
             }
         }
+        return childPids;
+    }
+
+    private static Set<String> getChildrenOrDescendants(String parentPid, boolean allDescendants) {
+        Set<String> descendantPids = new HashSet<>();
```

Comment:
You don' need to create a queue just to add the results of `getChildren()`.  Just return that set directly in the case where `allDescendants` is false.  SO this whole method would be simply:
```
// note change of argument to int
private static Set<String> getChildrenOrDescendants(int parentPid, boolean allDescendants) {
    if (allDescendants) {
        // no need to parse parentPid to int in this call
        return getChildrenOrDescendants( ... );
    }
    // change this to take an int argument
    return getChildren(parentPid);
}
```

Reply:
Actually it's even simpler.  Just call `getChildren` from `queryChildProcesses()` and call `getChildrenOrDescendants()` directly from `queryDescendantProcesses()`. No need for this intermediate method to combine the approaches and split them out again, so it can be deleted.